### PR TITLE
Hw06

### PR DESCRIPTION
**Pull request by ruslan-gorod**

### CODE REVIEW

File: src/com/ua/mate/hw06/HomeWork06.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        integerSet2.add(4);
+        integerSet2.add(7);
+        integerSet2.add(1);
+        integerSet2.add(5);
+        Set<Integer> integerSetResult = symmetricDifference(integerSet1, integerSet2);
+        System.out.println(Arrays.toString(integerSetResult.toArray()));
+    }
+
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        SortedSet<T> result = new TreeSet<>(new HashSet<>());
```
Comment (Siril-O) [inline src/com/ua/mate/hw06/HomeWork06.java]: why do you pass new HashSet<> to constructor

File: src/com/ua/mate/hw06/Animal.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                "name='" + name + '\'' +
+                '}';
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name);
+    }
+
+    public static Animal[] deserializeAnimalArray(byte[] data) {
```
Comment (Siril-O) [inline src/com/ua/mate/hw06/Animal.java]: Add test for this method

File: src/com/ua/mate/hw06/HomeWork06.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        Set<Integer> integerSetResult = symmetricDifference(integerSet1, integerSet2);
+        System.out.println(Arrays.toString(integerSetResult.toArray()));
+    }
+
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        SortedSet<T> result = new TreeSet<>();
+        Set<T> tmpSet = new HashSet<>(set1);
+        tmpSet.removeAll(set2);
+        result.addAll(tmpSet);
+        set2.removeAll(set1);
```
Comment (Siril-O) [inline src/com/ua/mate/hw06/HomeWork06.java]: you should not modify input arguments set1 set2

File: src/com/ua/mate/hw06/HomeWork06.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        integerSet2.add(4);
+        integerSet2.add(7);
+        integerSet2.add(1);
+        integerSet2.add(5);
+        Set<Integer> integerSetResult = symmetricDifference(integerSet1, integerSet2);
+        System.out.println(Arrays.toString(integerSetResult.toArray()));
+    }
+
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        SortedSet<T> result = new TreeSet<>();
```
Comment (Siril-O) [inline src/com/ua/mate/hw06/HomeWork06.java]: Why do you need TreeSet ?

Comment (Siril-O) [reply src/com/ua/mate/hw06/HomeWork06.java]: What interface should implement T to be valid to be used in TreeSet?

Comment (ruslan-gorod) [reply src/com/ua/mate/hw06/HomeWork06.java]: > Why do you need TreeSet ?
I want Set to be sorted

Comment (ruslan-gorod) [reply src/com/ua/mate/hw06/HomeWork06.java]: interface - Comparable

> What interface should implement T to be valid to be used in TreeSet?
Comparable

**Pull request merged**